Added documentation for the keba-charging-station component (#9616)
home-assistant/home-assistant.github.io: source/_components/keba.markdown

---
layout: page
title: "Keba Charging Station"
description: "Instructions on how to setup your Keba charging station with Home Assistant."
date: 2019-07-06 14:00
sidebar: true
comments: false
sharing: true
footer: true
logo: keba.png
ha_category:
  - Binary Sensor
  - Lock
  - Sensor

> ha_release: 0.96

Reviewer: ```suggestion
ha_release: 0.98
```

      required: false
      type: integer
      default: 30
    failsafe_fallback:
      description: Fallback current of the failsafe mode in A. Allowed values are between 6 Ampere and 63 Ampere. 0 Ampere disables the running charging process completely (this parameter is only used if failsafe mode is enabled).
      required: false
      type: integer
      default: 6
    failsafe_persist:
      description: Saving the failsafe configuration to internal EEPROM of the Keba charging station. 1 means save it, 0 means do only keep this configuration until the next restart of the charging station (this parameter is only used if failsafe mode is enabled).
      required: false
      type: integer
      default: 0
    refresh_interval:
      description: Refresh interval to fetch new data from the charging station. 5 seconds (same as in the official app) is recommended.
      required: false
      type: integer
      default: 5
{% endconfiguration %}

## Services

The `keba` component offers several services. Using these services will change the state of your charging station. So use these services with care!

### Authorizing and Deauthorizing `keba.authorize` and `keba.deauthorize`

> The charging station can be authorize and deauthorize vai service calls (`keba.authorize` and `keba.deauthorize`) or via the lock component that is created automatically for the charging station. In both cases the RFID tag from the configuration is used.

Reviewer: ```suggestion
The charging station can be authorized and deauthorized via service calls (`keba.authorize` and `keba.deauthorize`) or via the lock component that is created automatically for the charging station. In both cases the RFID tag from the configuration is used.
```

### Start and Stop `keba.start` and `keba.stop`

The service `keba.start` and `keba.stop` controls the charging process if the car is already authorized. Technically it sends `ena 1` or `ena 0` commands to the charging station.

### Set Target Energy `keba.set_energy`

The service `keba.set_energy` sets the target energy for the current session to the given energy attribute in kWh. Payload example:
```json
{
  "energy": 10.0
}
```

### Maximum Current `keba.set_curr`

The service `keba.set_curr` sets the maximum current to the given current attribute in Ampere. Payload example:
```json
{
  "current": 16.0
}
```

### Request New Data `keba.request_data`

> The service `keba.request_data` send data update requests to the charging station.

Reviewer: ```suggestion
The service `keba.request_data` sends data update requests to the charging station.
```

### Request New Data `keba.set_failsafe`

> The service `keba.set_failsafe` set the failsafe mode of the charging station. Payload example:

Reviewer: ```suggestion
The service `keba.set_failsafe` sets the failsafe mode of the charging station. Payload example:
```

---
layout: page
title: "Keba Charging Station"
description: "Instructions on how to setup your Keba charging station with Home Assistant."
date: 2019-06-11 20:00
sidebar: true
comments: false
sharing: true
footer: true
logo: keba.png
ha_category:
  - Binary Sensor
  - Lock
  - Sensor

> ha_release: 0.94

Reviewer 2: ```suggestion
ha_release: 0.95
```

> redirect_from:

Reviewer 2: Please remove the `redirect_from`

Author: Fixed, I probably used an old style documentation as template.

Reviewer 2: Well what I mainly see is that people unquestionably take over without really knowing what it is for 😉

      required: false
      type: string
      default: "00845500"
    failsafe:
      description: Enable failsafe mode at home assistant startup.
      required: false
      type: boolean
      default: false
    failsafe_timeout:
      description: Timeout of the failsafe mode in seconds. Allowed values are between 10 seconds and 600 seconds (only if failsafe mode is enabled). Make sure to call the `keba.set_curr` service regularly within this timeout period!
      required: false
      type: integer
      default: 30
    failsafe_fallback:
      description: Fallback current of the failsafe mode in A. Allowed values are between 6 Ampere and 63 Ampere. 0 Ampere disables the running charging process completely (only if failsafe mode is enabled).
      required: false
      type: integer
      default: 6
    failsafe_save:
      description: Saving the failsafe configuration to internal EEPROM of the Keba charging station. 1 means save it, 0 means do only keep this configuration until the next restart of the charging station (only if failsafe mode is enabled).
      required: false
      type: integer
      default: 0
{% endconfiguration %}

> ## {% linkable_title Services %}

Reviewer 2: Maybe put the service names in the titles

Author: I am not 100% sure what your suggestion is, but I now added the service names in the descritpion of the service and additionally added some example payloads to the services calls with parameters.
